docs/guides/codex-app-server-trace.md

```
@@ -0,0 +1,89 @@
+| `mcpServer/startupStatus/updated` | MCP server boot progress | early, repeats | A:3,4 / B:3,4 |
+| `thread/status/changed` | Thread state transition (idle ↔ running) | brackets each turn | A:5,17 / B:5,35 |
+| `turn/started` | Turn began | first event of turn | A:6 / B:6 |
+| `item/started` (userMessage) | Codex echoes the user's input as an item | top of turn | A:7 / B:7 |
+| `item/completed` (userMessage) | Echo finalized | immediately after | A:8 / B:8 |
+| `account/rateLimits/updated` | Quota snapshot | mid-turn, multiple | A:9,16 / B:9,11,34 |
+| `thread/tokenUsage/updated` | Streaming usage refresh | mid-turn | B:10,33 |
+| `item/started` (reasoning) | Reasoning item opened | before tool calls / message | A:10 |
+| `item/completed` (reasoning) | Reasoning closed with `summaryLen: 0` | A:11 (no text payload) | A:11 |
+| `item/started` (commandExecution) | Shell command starting | between reasoning and final message | B:12 |
+| `item/commandExecution/outputDelta` | Streaming stdout/stderr token | repeated | B:13,14 |
+| `item/completed` (commandExecution) | Command finished, includes `command`, `aggregatedOutput`, `exitCode` | B:15 |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 4; Actual: 3; Too few cells, row will be missing data

(MD056, table-column-count)